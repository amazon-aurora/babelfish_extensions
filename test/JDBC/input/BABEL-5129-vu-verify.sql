-- Tests for ISNUMERIC function with varchar and nvarchar variables

SELECT * FROM babel_5129
GO
-- Test int
SELECT ISNUMERIC(int_type)
FROM babel_5129
GO
-- Test numeric
SELECT ISNUMERIC(numeric_type)
FROM babel_5129
GO
-- Test money
SELECT ISNUMERIC(money_type)
FROM babel_5129
GO
-- Test varchar
SELECT ISNUMERIC(varchar_type)
FROM babel_5129
GO
-- Test nvarchar
SELECT ISNUMERIC(nvarchar_type)
FROM babel_5129
GO

-- Test numeric variable
DECLARE @a numeric(24,6);
SELECT @a = 12.3420000000;
SELECT ISNUMERIC(@a), LEN(@a), DATALENGTH(@a)
GO

-- Test varchar variable
DECLARE @v varchar(20);
SELECT @v = '12.3420000000';
SELECT ISNUMERIC(@v), LEN(@v), DATALENGTH(@v)
GO

-- Test nvarchar variable
DECLARE @nv nvarchar(10);
SELECT @nv = '12.3420000000';
SELECT ISNUMERIC(@nv), LEN(@nv), DATALENGTH(@nv)
GO

-- Test empty varchar variable
DECLARE @v varchar(20);
SELECT @v = '';
SELECT ISNUMERIC(@v), LEN(@v), DATALENGTH(@v)
GO

-- Test empty nvarchar variable
DECLARE @nv nvarchar(10);
SELECT @nv = '';
SELECT ISNUMERIC(@nv), LEN(@nv), DATALENGTH(@nv)
GO

-- Test varchar variable with invalid numeric
DECLARE @v varchar(20);
SELECT @v = '12.34.20000000';
SELECT ISNUMERIC(@v), LEN(@v), DATALENGTH(@v)
GO

-- Test nvarchar variable with invalid numeric
DECLARE @nv nvarchar(10);
SELECT @nv = '12.34.20000000';
SELECT ISNUMERIC(@nv), LEN(@nv), DATALENGTH(@nv)
GO