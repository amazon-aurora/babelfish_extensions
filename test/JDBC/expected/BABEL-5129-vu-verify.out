
-- Tests for ISNUMERIC function with varchar and nvarchar variables
SELECT * FROM babel_5129
GO
~~START~~
int#!#numeric#!#money#!#varchar#!#nvarchar
45000#!#12345.12000#!#237891.2200#!#12.3420000000#!#12.3420000000
~~END~~

-- Test int
SELECT ISNUMERIC(int_type)
FROM babel_5129
GO
~~START~~
int
1
~~END~~

-- Test numeric
SELECT ISNUMERIC(numeric_type)
FROM babel_5129
GO
~~START~~
int
1
~~END~~

-- Test money
SELECT ISNUMERIC(money_type)
FROM babel_5129
GO
~~START~~
int
1
~~END~~

-- Test varchar
SELECT ISNUMERIC(varchar_type)
FROM babel_5129
GO
~~START~~
int
1
~~END~~

-- Test nvarchar
SELECT ISNUMERIC(nvarchar_type)
FROM babel_5129
GO
~~START~~
int
1
~~END~~


-- Test numeric variable
DECLARE @a numeric(24,6);
SELECT @a = 12.3420000000;
SELECT ISNUMERIC(@a), LEN(@a), DATALENGTH(@a)
GO
~~START~~
int#!#int#!#int
1#!#9#!#6
~~END~~


-- Test varchar variable
DECLARE @v varchar(20);
SELECT @v = '12.3420000000';
SELECT ISNUMERIC(@v), LEN(@v), DATALENGTH(@v)
GO
~~START~~
int#!#int#!#int
1#!#13#!#13
~~END~~


-- Test nvarchar variable
DECLARE @nv nvarchar(10);
SELECT @nv = '12.3420000000';
SELECT ISNUMERIC(@nv), LEN(@nv), DATALENGTH(@nv)
GO
~~START~~
int#!#int#!#int
1#!#10#!#10
~~END~~


-- Test empty varchar variable
DECLARE @v varchar(20);
SELECT @v = '';
SELECT ISNUMERIC(@v), LEN(@v), DATALENGTH(@v)
GO
~~START~~
int#!#int#!#int
0#!#0#!#0
~~END~~


-- Test empty nvarchar variable
DECLARE @nv nvarchar(10);
SELECT @nv = '';
SELECT ISNUMERIC(@nv), LEN(@nv), DATALENGTH(@nv)
GO
~~START~~
int#!#int#!#int
0#!#0#!#0
~~END~~


-- Test varchar variable with invalid numeric
DECLARE @v varchar(20);
SELECT @v = '12.34.20000000';
SELECT ISNUMERIC(@v), LEN(@v), DATALENGTH(@v)
GO
~~START~~
int#!#int#!#int
0#!#14#!#14
~~END~~


-- Test nvarchar variable with invalid numeric
DECLARE @nv nvarchar(10);
SELECT @nv = '12.34.20000000';
SELECT ISNUMERIC(@nv), LEN(@nv), DATALENGTH(@nv)
GO
~~START~~
int#!#int#!#int
0#!#10#!#10
~~END~~

